dali/kernels/test/kernel_poc_test.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2018-2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <random>
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/test/test_tensors.h"
+#include "dali/kernels/tensor_shape_str.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+
+using std::cout;
+using std::cerr;
+using std::endl;
+
+namespace dali {
+namespace kernels {
+
+template <typename Input1, typename Input2, typename Output>
+struct MADKernel {
```

Comment:
what is MAD standing for here?

Reply:
MultiplyAdd. Not calling it FMA because per-element operation is written as `a*x+y`, not `fma(x, a, y)`.

Reply:
Maybe add some one liner comment above for that?

Reply:
Sure, will do.